File: app/src/main/java/in/woloo/www/trendingblog/epoxy/CarouselIndicatorModel.java
package in.woloo.www.trendingblog.epoxy;

import android.content.Context;

import androidx.recyclerview.widget.RecyclerView;

import com.airbnb.epoxy.Carousel;
import com.airbnb.epoxy.CarouselModel_;
import com.airbnb.epoxy.ModelView;
import com.google.android.material.progressindicator.CircularProgressIndicator;

import in.woloo.www.customview.CircularPagerIndicator;
import in.woloo.www.utils.Logger;

public class CarouselIndicatorModel extends CarouselModel_ {
    private final CircularPagerIndicator indicator = new CircularPagerIndicator();

    @Override
    public void bind(Carousel carousel) {
        super.bind(carousel);
        carousel.addItemDecoration(indicator);
        Logger.i("MODELS", "" + models().size());
    }

    @Override
    public void unbind(Carousel carousel) {
        super.unbind(carousel);
        carousel.removeItemDecoration(indicator);
    }


}

File: app/src/main/java/in/woloo/www/more/trendingblog/epoxy/CarouselIndicatorModel.kt
package `in`.woloo.www.more.trendingblog.epoxy

import androidx.core.content.ContextCompat
import com.airbnb.epoxy.Carousel
import com.airbnb.epoxy.CarouselModel_
import `in`.woloo.www.R
import `in`.woloo.www.application_kotlin.utilities.CircularPagerIndicator
import `in`.woloo.www.utils.Logger

class CarouselIndicatorModel : CarouselModel_() {
    private val indicator: CircularPagerIndicator = CircularPagerIndicator()
    override fun bind(carousel: Carousel) {
        super.bind(carousel)
        carousel.addItemDecoration(indicator)
        Logger.i("MODELS", "" + models().size)
        carousel.setBackgroundColor(
            ContextCompat.getColor(
                carousel.context,
                R.color.red
            )
        ) // Replace 'your_color' with your color resource
    }

    override fun unbind(carousel: Carousel) {
        super.unbind(carousel)
        carousel.removeItemDecoration(indicator)
    }
}
